pxblue/react-native-component-library
# Implement the full CollapsibleHeaderLayout

For PXBLUE-XXXX


#### Changes proposed in this Pull Request:
- Enhance the Header component to manage 'dynamic' or 'static' variants
- Implement the CollapsibleHeaderLayout component which manages the spacing and scrolling of the main screen contents
- Update the docs / storybook
- Did some linting cleanup on hooks/dependencies


#### Screenshots:
![collapsibleHeaderLayout](https://user-images.githubusercontent.com/29152776/122318332-dec52d80-ceec-11eb-9d76-b4e34bb26b6f.gif)



#### To Test:
- Swap the outer View component in the Showcase App.tsx with CollapsibleHeaderLayout
- Take the props from the existing Header and pass them into the `HeaderProps` prop of the CollapsibleHeaderLayout
- Remove the existing Header component
- Explore

> I have not checked this on Android yet, so that would be as good a place to start as any.

Alternatively, you can use the version of App.tsx I will share on Teams which does the above for you.
- The two menu buttons in the actionItems will swap the variant (first) and update the expanded/collapsed heights (second) if you want to test those out.


Make sure to poke around with all of the different combinations of things you can do (scrolled down, scrolled up, expanding and collapsing on click, opening and closing search in different locations, changing the sizes, changing the variant, etc.).
